src/main/webapp/script.js

```diff
@@ -41,47 +42,69 @@ function initCommentSlider() {
  * @return {JSON} comments json object of comment objects
  */
 async function getYouTubeComments(url) {
-  url = extractYouTubeUrl(url);
-  const numComments = document.getElementById(SLIDER_NAME).value;
-  const response = await fetch('/YouTubeComments?url=' + url +
-    '&numComments=' + numComments);
-  const comments = await response.json();
-  return comments;
+  try {
+    url = extractYouTubeUrl(url);
+    const numComments = document.getElementById(SLIDER_NAME).value;
+    const response = await fetch('/YouTubeComments?url=' + url +
+      '&numComments=' + numComments);
+    const comments = await response.json();
+    return comments;
+  } catch (err) {
+    throw new Error('Error retrieving comments.');
+  }
 }
 
 /**
  * Wrapper function for preparing onClick function
  */
 function onButtonPress() {
   $('#submit-link-btn').click(function() {
-    showLoadingGif();  
-    const urlInput = document.getElementById('link-input').value;
-    updateUIWithVideoContext(urlInput);
-    displayOverallResults(urlInput);
+    try {
+      toggleErrorOutput('none');
+      showLoadingGif();
+      const urlInput = document.getElementById('link-input').value;
+      updateUIWithVideoContext(urlInput);
+      displayOverallResults(urlInput);
+    } catch (err) {
+      displayError(err.message);
+    }
   });
 }
 
+// Toggles visibility of error message
+function toggleErrorOutput(mode) {
+  document.getElementById(ERROR_OUTPUT_ID).style.display = mode;
+}
+
+function displayError(message) {
```

Comment:
Let's change this method contract to optionally take an error.

We want to ensure that if any errors are encountered, they are logged to the console, and we could do this easily if we centralize this error logging logic within this function.